src/main/java/org/broadinstitute/hellbender/tools/spark/sv/BreakpointPair.java

```diff
@@ -0,0 +1,476 @@
+        double err = Double.MAX_VALUE;
+        for(int cn2=1; cn2<MAX_LOWER_CN; ++cn2) {
+            for(int cn1=cn2+1; cn1<=2*cn2; ++cn1) {
+                final int dupLenUpperBound = cn1==2*cn2 ? overlapOnLowerCNSequence : overlapOnHigherCNSequence;
+                for (int l=2; l<=dupLenUpperBound;++l) {
+                    for (int lambda=0; lambda<l; ++lambda) {
+                        final int d1 = (2*cn2 - cn1)*l + lambda;
+                        final int d2 = cn2*l + lambda;
+                        final double newErr = Math.abs(overlapOnHigherCNSequence-d1) + Math.abs(overlapOnLowerCNSequence-d2);
+                        if(newErr<err){
+                            err = newErr;
+                            higherCnEst = cn1; lowerCnEst = cn2;
+                            unitLen= l; pseudoHomLen = lambda;
+                        }
+                        if (err<1) return new int[]{lowerCnEst, higherCnEst, unitLen, pseudoHomLen};
+                    }
+                }
+            }
+        }
+
+        return new int[]{lowerCnEst, higherCnEst, unitLen, pseudoHomLen};
+    }
+
+    // TODO: 12/11/16 again this does not deal with inter-chromosome translocation yet
+    /**
+     * Returns the reference coordinates of the left and right breakpoints implied by this chimeric alignment.
+     * If there is homologous sequence represented in the alignments, it will be assigned to the side of the breakpoint
+     * with higher reference coordinates.
+     */
+    @VisibleForTesting
+    static Tuple2<SimpleInterval, SimpleInterval> leftJustifyBreakpoints(final ChimericAlignment ca, final BreakpointComplication complication) {
+
+        final int homologyLen = complication.homologyForwardStrandRep.length();
+
+        final String leftBreakpointRefContig, rightBreakpointRefContig;
+        final int leftBreakpointCoord, rightBreakpointCoord;
+        if(complication.refDup!=0 && complication.contigDup!=0) { // todo : development artifact-- assuming tandem duplication is not co-existing with inversion
+            final int dupLen = complication.dupSeqForwardStrandRep.length();
+            final String contig = ca.regionWithLowerCoordOnContig.referenceInterval.getContig();
+            leftBreakpointRefContig = rightBreakpointRefContig = contig;
+            final SimpleInterval leftReferenceInterval, rightReferenceInterval;
+            if (ca.isForwardStrandRepresentation) {
+                leftReferenceInterval = ca.regionWithLowerCoordOnContig.referenceInterval;
+                rightReferenceInterval = ca.regionWithHigherCoordOnContig.referenceInterval;
+            } else {
+                leftReferenceInterval = ca.regionWithHigherCoordOnContig.referenceInterval;
+                rightReferenceInterval = ca.regionWithLowerCoordOnContig.referenceInterval;
+            }
+            leftBreakpointCoord = leftReferenceInterval.getEnd() - homologyLen - Math.max(0, complication.contigDup-complication.refDup)*dupLen;
```

Comment:
Yes, let's talk in person as I spent sometime to figure this out and should explain to you in person.